src/main/java/com/spotify/futures/FuturesExtra.java

```diff
@@ -323,6 +474,23 @@ public Z apply(List<Object> results) {
     Z apply(A a, B b, C c, D d, E e, F f);
   }
 
+  /**
+   * Asynchronously transform the result of 3 {@link ListenableFuture}s a, b, c, d, e and f once
+   * both has completed using {@code function} and complete the returned ListenableFuture at the
+   * point.
+   * Execution is asynchronous in the sense that function returns a Future that may be executed
+   * asynchronously and once it completes it will complete the future that this method returns.
+   *
+   * @param a the first ListenableFuture
+   * @param b the second ListenableFuture
+   * @param c the third ListenableFuture
+   * @param d the fourth ListenableFuture
+   * @param e the fifth ListenableFuture
+   * @param f the sixth ListenableFuture
+   * @param function an AsyncFunction6 implementation, possibly expressed as a java 8 lambda.
+   * @return a ListenableFuture of the result of {@code function}, that completes as soon as
+   * the future that gets returned by the supplied Function6.apply() has completed.
```

Comment:
Example:
LF complete = immediateFuture("complete");
LF result = FuturesExtra.asyncTransform2(futureA, futureB, (a, b) -> complete);

result is NOT completed as soon as the future returned by the lambda is completed.
I think we should avoid specifying the exact semantics - it's much easier to describe it as a wrapper around the actual transform function.